android/app/src/main/java/com/microsoft/reacttestapp/MainActivity.kt

```diff
@@ -72,27 +75,60 @@ class MainActivity : ReactActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
 
```

Comment:
You need to restore the value of `didInitialNavigation` from the `savedInstanceState` bundle here.